sftp/sftp.go

```diff
@@ -0,0 +1,97 @@
+package sftp
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	// PasswordAuth indicates password-based authentication
+	PasswordAuth = "passwordAuth"
+	// KeyAuth indicates key-based authentication
+	KeyAuth = "keyAuth"
+)
+
+// SFTPManager interface abstracts the SFTP client
+type SFTPManager interface {
+	UploadFile(localFilePath, remoteDir string) error
+	DownloadFile(remoteFilePath, localDir string) error
+	DeleteFile(remoteFilePath string) error
+}
+
+// SFTPManagerImpl is a real implementation of SFTPManager
+type SFTPManagerImpl struct {
+	client SFTPClient
+}
+
+func NewSFTPManager(sshClient *ssh.Client) (SFTPManager, error) {
+	sftpClient, err := NewSFTPClient(sshClient)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create SFTP client: %w", err)
+	}
+	return &SFTPManagerImpl{client: sftpClient}, nil
+}
+
+// UploadFile uploads a file to the remote server
+func (r *SFTPManagerImpl) UploadFile(localFilePath, remoteDir string) error {
```

Comment:
These methods should return errors with more context instead of just doing `return err`.

Example:

```go
localFile, err := os.Open(localFilePath)
if err != nil {
	return fmt.Errorf("cannot open file: %w", err)
}
```